test/libsolidity/syntaxTests/controlFlow/localStorageVariables/modifier_declaration_fine.sol

```diff
@@ -18,3 +18,6 @@ contract C {
 
 }
 // ----
+// Warning 5740: (247-275): Unreachable code.
+// Warning 5740: (328-332): Unreachable code.
+// Warning 5740: (348-366): Unreachable code.
```

Comment:
What exactly is unreachable here and why?

Reply:
```
Warning: Unreachable code.
  --> test/libsolidity/syntaxTests/controlFlow/localStorageVariables/modifier_declaration_fine.sol:12:60:
   |
12 |     function f(bool flag) revertIfNoReturn() internal view {
   |                                                            ^ (Relevant source part starts here and spans across multiple lines).

Warning: Unreachable code.
  --> test/libsolidity/syntaxTests/controlFlow/localStorageVariables/modifier_declaration_fine.sol:15:53:
   |
15 |     function g(bool flag) revertIfNoReturn() ifFlag(flag) internal view {
   |                                                     ^^^^

Warning: Unreachable code.
  --> test/libsolidity/syntaxTests/controlFlow/localStorageVariables/modifier_declaration_fine.sol:15:73:
   |
15 |     function g(bool flag) revertIfNoReturn() ifFlag(flag) internal view {
   |                                                                         ^ (Relevant source part starts here and spans across multiple lines).


```

Reply:
Basically, because the mod always reverts, nothing in the function is reachable. Though I realize it's not strictly correct because we do call the function, we just revert() after. If the revert undoes everything the function did, it is as if it can never actually run.. but I am going to look into this properly